### PR TITLE
Feat/redux

### PR DESCRIPTION
**Pull request by sievertknorrblad**

### CODE REVIEW

File: src/components/header/index.js [hunk 1]
```diff
@@ -8,11 +10,18 @@ const mainMenu = [
...
   ];
 
-export const Header = () => (
+const Header = ({ toggleTheme }) => (
   <div className="header">
     <Link to="/">
       <img src="https://news.ycombinator.com/y18.gif" alt="durov" />
     </Link>
     {mainMenu.map(item => (<a key={item.text} href={item.url}>{item.text}  </a>))}
+    <button onClick={e => toggleTheme()}>toggle the theme</button>
```
Comment (iamstarkov) [inline src/components/header/index.js:19]: do you know if you are using toggleTheme from props or imported one?

Comment (iamstarkov) [reply src/components/header/index.js:19]: its called variable overshadowing

Comment (iamstarkov) [reply src/components/header/index.js:19]: do this `import * as actions from '../../actions'` as I specified in the gist https://gist.github.com/iamstarkov/45c4fd7d9cc99b943b277995ad2f9f71#5-structuring-redux-related-code

File: src/actions/index.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      .getItem(id)
+      .then(item => {
+        dispatch(requestItemSuccess(item));
+      })
+      .catch(err => {
+        dispatch(requestItemFailure(id, err));
+      });
+  };
+
+  export { requestItemStart, requestItemSuccess, requestItemsIdsFailure, requestItemsIdsStart, requestItemsIdsSuccess, requestItemFailure, toggleTheme, updateItemsToShow };
```
Comment (iamstarkov) [inline src/actions/index.js]: you dont need to have all of these, just export every action and action creator on its own

File: src/components/newsItem/index.js [hunk 3]
```diff
@@ -1,60 +1,55 @@
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
 import { Link } from 'react-router-dom';
 import { distanceInWordsToNow } from 'date-fns';
 import './styles.css'
-import { api } from '../../utils';
+import { fetchItem as fetchItemAction } from '../../actions';
```
Comment (iamstarkov) [inline src/components/newsItem/index.js]: fix this as I mentioned before

File: src/components/newsItem/index.js [hunk 4]
```diff
@@ -1,60 +1,55 @@
...
-        )
-      })
+    const { fetchItem, id } = this.props
+    fetchItem(id)
   }
 
-  shouldComponentUpdate(nextProps, nextState) {
-    return this.state.item !== nextState.item
+  shouldComponentUpdate(nextProps) {
+    return this.props.item !== nextProps.item
```
Comment (iamstarkov) [inline src/components/newsItem/index.js]: what is your intention here?

File: src/components/newsItemList/index.js [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import { NewsItem } from '../newsItem';
+import { default as NewsItem } from '../newsItem';
```
Comment (iamstarkov) [inline src/components/newsItemList/index.js]: nope, refer to components from components index file: `import { NewsItem } from '../'`

File: src/components/pageNewsList/index.js [hunk 6]
```diff
@@ -1,42 +1,38 @@
 import React, { Component } from 'react';
-import { api } from '../../utils';
+import { connect } from 'react-redux';
 import { NewsItemList } from '../newsItemList';
```
Comment (iamstarkov) [inline src/components/pageNewsList/index.js]: and here

File: src/index.js [hunk 7]
```diff
@@ -3,6 +3,18 @@ import ReactDOM from 'react-dom';
...
-ReactDOM.render(<App />, document.getElementById('root'));
+const store = createStore(
+    rootReducer,
+    compose(applyMiddleware(thunk),
+            window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__())
+);
+ReactDOM.render(<Provider store={store}>
+                    <App />
+                </Provider>,
+                document.getElementById('root'));
```
Comment (iamstarkov) [inline src/index.js]: you have a bit odd indentation here, but its fine

File: src/components/index.js [hunk 8]
```diff
@@ -1,7 +1,7 @@
...
+export { default as Page } from './page';
+export { default as Header } from './header';
 export { Content } from './content';
 export { Footer } from './footer';
 export { NewsItemList } from './newsItemList';
-export { NewsItem } from './newsItem';
-export { PageNewsList } from './pageNewsList';
\ No newline at end of file
+export { default as NewsItem } from './newsItem';
+export { default as PageNewsList } from './pageNewsList';
```
Comment (iamstarkov) [inline src/components/index.js:7]: like you changed what you export to consume here in the. so please import from this file, not point to specific component folder

**Review by iamstarkov — CHANGES_REQUESTED** [review]
needs some work

File: src/components/newsItem/index.js (hunk 4 above)
Comment (iamstarkov) [reply src/components/newsItem/index.js]: I mean
```
> {} === {}
false
```

Comment (sievertknorrblad) [reply src/components/newsItem/index.js]: Can probably be removed right? We always want to re-render if we get new data in the state for this id?

File: src/index.js (hunk 7 above)
Comment (sievertknorrblad) [reply src/index.js]: Didn't know what else to do :P The 2 arguments passed should be aligned, and <App /> should be indented under <Provider> right :)?

File: src/components/newsItem/index.js (hunk 4 above)
Comment (iamstarkov) [reply src/components/newsItem/index.js]: yes, we can remove it

File: src/components/newsItemList/index.js [hunk 9]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import { NewsItem } from '../newsItem';
+import { NewsItem } from '../../components';
```
Comment (iamstarkov) [inline src/components/newsItemList/index.js]: how about `from '../'`?

**Review by iamstarkov — APPROVED** [review]
some minor comments, otherwise looks really good

**Review by iamstarkov — APPROVED** [review]
LGTM

**Pull request merged**